src/util/xeval/builtin_cast.rs

```diff
@@ -0,0 +1,337 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::i64;
+use tipb::expression::Expr;
+use util::codec::datum::Datum;
+use util::codec::mysql::{Decimal, Time, Duration, DEFAULT_FSP};
+use util::codec::convert;
+use super::{Evaluator, EvalContext, Result, Error};
+
+const ERROR_UNIMPLEMENTED: &'static str = "unimplemented";
+const TYPE_INT: &'static str = "int";
+
+fn invalid_type_error(datum: &Datum, expected_type: &str) -> Result<Datum> {
+    Err(Error::Eval(format!("invalid expr type: {:?}, expect: {}", datum, expected_type)))
+}
+
+fn check_datum_int(datum: &Datum) -> Result<()> {
+    match *datum {
+        Datum::I64(_) | Datum::U64(_) => Ok(()),
+        _ => Err(Error::Eval(format!("invalid expr type: {:?}, expect to be int", datum))),
+    }
+}
+
+impl Evaluator {
+    pub fn cast_int_as_int(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
```

Comment:
In TiDB,  `IntBuiltinFunc.eval()` will check `hasUnsignedFlag(builtinFunc.tp.Flag)`
If true, we set store the result as uint64 in datum.

https://github.com/pingcap/tidb/blob/master/expression/builtin.go#L266
https://github.com/pingcap/tidb/blob/master/expression/scalar_function.go#L175